Support for enum of boolean

darcyparker opened the pull request:
See https://github.com/bcherny/json-schema-to-typescript/issues/34

darcyparker commented on src/index.ts:
```diff
@@ -248,21 +251,23 @@ class Compiler {
 
   private validateEnumMembers(rule: JSONSchema.Schema): EnumType {
-    if (!rule.type) rule.type = 'string'
+    if (rule.type === undefined) rule.type = 'string'
```
Note: `!rule.type` returns `false` if `rule.type === 'boolean'` or any string value. I think the correct intent is `rule.type === undefined`

darcyparker replied:
Note: `!rule.type` returns `false` if `rule.type === 'boolean'` or any string value. I think the correct intent is `rule.type === undefined`

donaldpipowitch replied:
I don't think this is correct. It is valid to _not_ set a type here. If we default to `'string'` and our enum is `[ false ]`, this would be incorrect.

darcyparker replied:
if `"type" : "boolean"`,  it will be defined, so `rule.type` won't default to `string`.

donaldpipowitch replied:
Yes, _if_ `"type" : "boolean"` is defined. But if it isn't defined, it should _not_ default to string. This is how JSON schema works as far as I know.

darcyparker commented:
I took a crack at this because I thought the case of a type that is a subset of `boolean` was useful after reading @donaldpipowitch example.  I hope I interpreted the problem correctly. I am open to feedback or scrapping this I misunderstood the problem.

donaldpipowitch commented on test/cases/enumBoolean.ts:
```diff
@@ -0,0 +1,23 @@
+  "type": "object",
+  "properties": {
+      "foo": {
+        "type" : "boolean",
```
Maybe we could have another test _without_ a `"type": "boolean"`. Like here: https://github.com/webpack/webpack/blob/master/schemas/webpackOptionsSchema.json#L638

darcyparker replied:
Sure - that makes sense.  The code will break without it as it stands now. But I think it can be refactored to work.

darcyparker commented:
@donaldpipowitch and @bcherny: I refactored based on feedback and rebased. Let me know what you think now.

Note: As I started to add the new enumValidation tests cases, I found it confusing to navigate.  So I numbered them to correspond to the 6 validations in the order they are defined in validateEnumMembers(). The 3rd validation has 3 cases, so they are numbered `3.1`, `3.2`, `3.3`.

darcyparker commented:
Also note: I added `.nvmrc` to make it convenient to use the same node version that you are using for integration tests.  I use `avn` with `nvm` so that when I `cd` into a directory, my environment is using the right version of node.

darcyparker commented:
FYI: Just rebased on top of latest master branch.

bcherny commented on test/cases/enum.ts:
```diff
@@ -2,46 +2,58 @@ export var schema = {
+      '  anIntegerEnum: AnIntegerEnum;',
+      '  anImpliedIntegerEnum: AnImpliedIntegerEnum;',
+      '}'
+      ].join('\n')
```
Can you just use a single template string?

darcyparker commented:
Thanks for accepting this @bcherny.  And thanks for taking care of the single template string. (I thought the indentation looked better... but it's just a styling opinion. I am fine with adapting.)

BTW: I am glad to see your repo improving.  It won't be long before I am using it my projects.  It's going to really help reduce the maintenance of and improve the quality of my front end to back end interfaces/coupling.
